libnd4j/standalone/LabelsImagenet.h

```diff
@@ -0,0 +1,1014 @@
+#ifndef LABELS_IMAGENET_H
```

Comment:
And here as well. Although I'm not sure we can claim copyright on the list of labels. Let's just load it from a plain text file given by the user? It also makes the tool more useful that way.

Reply:
I think this tool would be useless if we did not provide the labels somehow so I dug a bit more in details. When I included it first, I checked quickly and saw that many people included that in various project, as if it was public domain, slapped several permissible licenses on their own reformatted list.

It turns out that it comes originally from ImageNet's Large Scale Visual Recognition Challenge 2014 (ILSVRC2014), who does not state a license explicitly but says in its rules (http://www.image-net.org/challenges/LSVRC/2014/index#cite)
"If you are reporting results of the challenge or using the dataset, please cite:

    Olga Russakovsky*, Jia Deng*, Hao Su, Jonathan Krause, Sanjeev Satheesh, Sean Ma, Zhiheng Huang, Andrej Karpathy, Aditya Khosla, Michael Bernstein, Alexander C. Berg and Li Fei-Fei. (* = equal contribution) ImageNet Large Scale Visual Recognition Challenge. IJCV, 2015."

Which is this paper: https://arxiv.org/abs/1409.0575, which contains a list of all the categories used. I could not find any other licensing information about the ILSVRC2014 dataset. 

But wait, there is more. What the ILSVRC2014 actually gives is not a list of strings. It is a list of references to WordNet synset (a set of synonyms). So the actual strings come from WordNet, which is under this license (https://wordnet.princeton.edu/license-and-commercial-use) that both wikipedia and the FSF classify as a form of BSD 2-clauses licenses (which on cursory glance seems correct)

Technically I would argue that this file is more documentation/reference than code. It should probably be distributed alongside the models.

Reply:
Added a header for now. I can't work on it today but I'll write a loader later. Added on the todo list.

Reply:
Yes, it's specific to each model, so that's why we should just have the tool load it from a plain text file anyway. It's available, for example, as synset_words.txt inside http://dl.caffe.berkeleyvision.org/caffe_ilsvrc12.tar.gz . I think we have it online on a server from Skymind somewhere too (@AlexDBlack ? @raver119 ?) because that way we don't have to go through Eclipse's IP team.

Reply:
@yquemener The list is available here:
https://github.com/deeplearning4j/dl4j-test-resources/blob/master/src/main/resources/tf_graphs/zoo_models/labels/imagenet_labellist.txt
So use that file for your tests.